bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/HandleFactory.java

```diff
@@ -24,7 +24,7 @@
 import java.io.IOException;
 
 interface HandleFactory {
-    LedgerDescriptor getHandle(long ledgerId, byte[] masterKey)
+    LedgerDescriptor getHandle(long ledgerId, byte[] masterKey, boolean journalReplay)
```

Comment:
will `readOnly` better than `journalReplay`?

Reply:
- Get a handle that is read-only will call `getReadOnlyHandle`
- There is only one scenario that will call `getHandle` to read, it is journal replay.